### DISCUSSION TITLE
Add Osu Mod to Force Full Alternate

### DISCUSSION THREAD

**Discussion by StayBlue** [ppy/osu#16479] [Ideas] [answered]
I know there have been some discussions and PRs in the past to add this mod like [this one](https://github.com/ppy/osu/pull/9606) and [this one](https://github.com/ppy/osu/discussions/13003) but these efforts seem to be dead now and I have been really eager for something like this to be implemented into lazer!

While the other discussions and PRs have explained the implementation/idea better than I can (see above for the links), I will try to summarize the mod. In essence, the mod would force the player to alternate between the two keys by locking one of the keys when they have been pressed until the other key is pressed and vice versa (basically, alternating).

I know the mod implementation isn't that easy, but it's one of those luxuries that would really make it or break it for me on whether to switch from osu!stable to lazer once the transition starts. It's especially useful for me, a Linux user, who doesn't have the luxury of simply opening up a macro that would do the same thing the proposed mod does but externally.

I think it would be a great addition to lazer and would allow players to play on lazer without having to go to practice clients like McOsu.
 
P.S. sorry for double-posting, but the discussion has been dead for a while now and I thought it would be worthwhile to respark the conversation.

**Comment by bdach** [comment, marked as answer]
there is no need for this discussion. the proper course of action would be to bump the existing discussion rather than open a new one.

and from what i remember the osu!-only implementation was correct, the osu!taiko one was the one with the bigger number of problems. if the osu!-only implementation was split off, then it could get merged quicker.

**Reply by StayBlue** [reply]
My bad, I'll try to remember that for the future. Is there any possibility of it being split up and merged any time soon?

**Reply by mk56-spn** [reply]
ill give it a shot after my FL change pr
